src/main.py

```diff
@@ -14,28 +27,54 @@ def is_controller_connected():
 
 
 """
-Script to determine leader and follower roles.
+This function calls the ./bt_toggle script with the specified
+command and address
+"""
+def bt_toggle(command, address):
+    subprocess.call(['./bt_toggle.sh', command, address])
 
-Takes input from the camera to determine if the current vehicle
-should lead or follow based on whether or not there exists an ARTag
-in front of the vehicle.
 
-If an ARTag exists, the vehicle will become a follower vehicle.
-If an ARTag does not exist, the vehicle will become the leader vehicle.
-"""
 def main():
     tag = TagRecognition()
+    bt_address = '5C:BA:37:26:6D:9A'
 
-    # If an ARTag is detected, there is a vehicle ahead of this vehicle so it will follow.
-    # If an ARTag is not detected, no vehicle is present so this vehicle will be a leader.
-    if tag.detect() is not None:
-        follower = Follower()
-        while True:
-            follower.follow()
+    # If an ARTag is detected, the vehicle will try to disconnect from a bluetooth controller and
+    # if it is not connected to one it will follow. If an ARTag is not detected, the vehicle will
+    # will try to connect to the bluetooth controller and if succeful, the vehicle will be a leader.
+    if tag.detect():
+        bt_toggle('disconnect', bt_address)
     else:
-        leader = Leader()
-        while True:
-            leader.lead()
+        bt_toggle('connect', bt_address)
+    if is_controller_connected():
+        vehicle = Leader()
+    else:
+        vehicle = Follower()
+
+    # This loop makes vehicle move. If the vehicle is a leader and sees an ARTag
```

Comment:
This is way too verbose. It takes longer to read this than it does to read the code. This should be like two lines.